protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -377,37 +418,38 @@ fn execute_cheatcode_hint(
                 .expect("Failed to get contract_address from return_data");
             let contract_address = Felt252::from_bytes_be(contract_address.bytes());
 
-            insert_value_to_cellref!(vm, deployed_contract_address, contract_address)?;
-            // todo in case of error, consider filling the panic data instead of packing in rust
-            insert_value_to_cellref!(vm, panic_data_start, Felt252::from(0))?;
-            insert_value_to_cellref!(vm, panic_data_end, Felt252::from(0))?;
-
-            Ok(())
+            // TODO: in case of error, consider filling the panic data instead of packing in rust
+            insert_at_pointer(vm, &mut result_segment_ptr, Felt252::from(0)).unwrap();
+            insert_at_pointer(vm, &mut result_segment_ptr, contract_address).unwrap();
         }
-        &ProtostarHint::Prepare { .. } => todo!(),
-        &ProtostarHint::Call { .. } => todo!(),
-        ProtostarHint::Print { start, end } => {
-            let as_relocatable = |vm, value| {
-                let (base, offset) = extract_buffer(value);
-                get_ptr(vm, base, &offset)
-            };
-
-            let mut curr = as_relocatable(vm, start)?;
-            let end = as_relocatable(vm, end)?;
-
-            while curr != end {
-                let value = vm.get_integer(curr)?;
-                if let Some(shortstring) = as_cairo_short_string(&value) {
-                    println!("original value: [{value}], converted to a string: [{shortstring}]",);
+        "print" => {
+            for value in inputs {
+                if let Some(short_string) = as_cairo_short_string(&value) {
+                    println!("original value: [{value}], converted to a string: [{short_string}]",);
                 } else {
                     println!("original value: [{value}]");
                 }
-                curr += 1;
             }
-
-            Ok(())
         }
+        _ => Err(HintError::CustomHint(Box::from(format!(
+            "Unknown cheatcode selector: {selector}"
+        ))))?,
     }
+
+    let result_end = result_segment_ptr;
+    insert_value_to_cellref!(vm, output_start, result_start)?;
+    insert_value_to_cellref!(vm, output_end, result_end)?;
+
+    Ok(())
+}
+
+// TODO: remove this when extract_relocatable is pub in cairo
+fn extract_relocatable(
+    vm: &VirtualMachine,
+    buffer: &ResOperand,
+) -> Result<Relocatable, VirtualMachineError> {
+    let (base, offset) = extract_buffer(buffer);
+    get_ptr(vm, base, &offset)
 }
```

Comment:
When extract_relocatable plans to be public in cairo? Do we have a PR/issue to change it?

Reply:
I asked @abulenok to make a few functions public in the Cairo PR (the one concerning `MemBuffer`) he is doing rn, this one is included